.github/workflows/python.yml

```diff
@@ -17,37 +17,33 @@ jobs:
     defaults:
       run:
         working-directory: python/core
-
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: "recursive"
-
       - name: Install Rust
         uses: dtolnay/rust-toolchain@stable
         with:
           components: rustfmt, clippy
-
       - uses: Swatinem/rust-cache@v2
-
-      - name: Cargo fmt
-        run: cargo fmt --all -- --check
-
-      # # Note not sure why this is needed
-      # - name: Install protobuf
-      #   run: |
-      #     sudo apt-get install libprotobuf-dev protobuf-compiler
-
-      # Don't use --all-features as we don't have GDAL 3.6 available in CI
-      - name: "clippy --all"
-        run: cargo clippy --all --tests -- -D warnings
-
-      - name: "cargo check"
-        run: cargo check --all
-
-      - name: "cargo test"
+      - name: Install Pixi
         run: |
-          cargo test --all
+          curl -fsSL https://pixi.sh/install.sh | bash
+          echo "$HOME/.pixi/bin" >> $GITHUB_PATH
```

Comment:
Maybe we should also consider using this: https://github.com/marketplace/actions/setup-pixi

Reply:
Then we can also easily opt in to `cache: true`

Reply:
I like that, I'll take a stab